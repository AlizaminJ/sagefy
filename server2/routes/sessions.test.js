const request = require('supertest')
const express = require('express')
const bodyParser = require('body-parser')

require('express-async-errors')

const sessionsRouter = require('./sessions')
const sessionMiddleware = require('../middleware/sessionMiddleware')
const errorMiddleware = require('../middleware/errorMiddleware')

describe('Sessions Routes', () => {
  const app = express()
  app.use(errorMiddleware)
  app.use(sessionMiddleware)
  app.use(bodyParser.json())
  app.use('/x/sessions', sessionsRouter)

  describe('GET /x/sessions', () => {
    test('get the session (logged out)', () =>
      request(app)
        .get('/x/sessions')
        .expect(res => expect(res.text).toMatchSnapshot())
        .expect(401)
        .expect('Content-Type', /json/))

    test('get the session (logged in)', () =>
      request(app)
        .post('/x/sessions')
        .send({ name: 'doris', password: 'example1' })
        .set('Accept', /application\/json/)
        .expect(200)
        .then(() =>
          request(app)
            .get('/x/sessions')
            .expect(res => expect(res.text).toMatchSnapshot())
            .expect(200)
            .expect('Content-Type', /json/)
        ))
  })

  // Order changed so that when we 'get' the session we're already logged in.
  describe('POST /x/sessions', () => {
    test('create a session (log in)', () =>
      request(app)
        .post('/x/sessions')
        .send({ name: 'doris', password: 'example1' })
        .set('Accept', /application\/json/)
        .expect(res => expect(res.text).toMatchSnapshot())
        .expect(200)
        .expect('Content-Type', /json/))
  })

  describe('PUT /x/sessions', () => {
    test('update a session -- choose a subject', () =>
      request(app)
        .put('/x/sessions')
        .send({ subject: 'd8212158-09e9-40b7-80af-4a0c014f42ba' })
        .set('Accept', /application\/json/)
        .expect(res => expect(res.text).toMatchSnapshot())
        .expect(200)
        .expect('Content-Type', /json/))

    test('update a session -- choose a unit', () =>
      request(app)
        .put('/x/sessions')
        .send({ unit: '1588ae27-86a6-4158-9d7d-2cf15c136b83' })
        .set('Accept', /application\/json/)
        .expect(res => expect(res.text).toMatchSnapshot())
        .expect(200)
        .expect('Content-Type', /json/))
  })

  describe('DELETE /x/sessions', () => {
    test('delete a session (log out)', () =>
      request(app)
        .del('/x/sessions')
        .set('Accept', /application\/json/)
        .expect(res => expect(res.text).toMatchSnapshot())
        .expect(200)
        .expect('Content-Type', /json/))
  })
})
