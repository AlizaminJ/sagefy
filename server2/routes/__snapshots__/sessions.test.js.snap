// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions Routes DELETE /x/sessions delete a session (log out) 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Cannot DELETE /x/sessions</pre>
</body>
</html>
"
`;

exports[`Sessions Routes GET /x/sessions get the session (logged out) 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Error: GlqI6VAROUCGqZy3OuKJ1A<br> &nbsp; &nbsp;at abort (/server/helpers/abort.js:4:22)<br> &nbsp; &nbsp;at abort (/server/middleware/authMiddleware.js:4:32)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/route.js:137:13)<br> &nbsp; &nbsp;at Route.dispatch (/server/node_modules/express/lib/router/route.js:112:3)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:281:22<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at Function.handle (/server/node_modules/express/lib/router/index.js:174:3)<br> &nbsp; &nbsp;at router (/server/node_modules/express/lib/router/index.js:47:12)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at jsonParser (/server/node_modules/body-parser/lib/types/json.js:110:7)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at session (/server/node_modules/express-session/index.js:454:7)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:91:12)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at expressInit (/server/node_modules/express/lib/middleware/init.js:40:5)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at query (/server/node_modules/express/lib/middleware/query.js:45:5)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at Function.handle (/server/node_modules/express/lib/router/index.js:174:3)<br> &nbsp; &nbsp;at Function.handle (/server/node_modules/express/lib/application.js:174:10)<br> &nbsp; &nbsp;at Server.app (/server/node_modules/express/lib/express.js:39:9)<br> &nbsp; &nbsp;at emitTwo (events.js:126:13)<br> &nbsp; &nbsp;at Server.emit (events.js:214:7)<br> &nbsp; &nbsp;at parserOnIncoming (_http_server.js:619:12)<br> &nbsp; &nbsp;at HTTPParser.parserOnHeadersComplete (_http_common.js:112:17)</pre>
</body>
</html>
"
`;

exports[`Sessions Routes POST /x/sessions create a session (log in) 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Error: Illegal arguments: string, undefined<br> &nbsp; &nbsp;at _async (/server/node_modules/bcryptjs/dist/bcrypt.js:286:46)<br> &nbsp; &nbsp;at /server/node_modules/bcryptjs/dist/bcrypt.js:307:17<br> &nbsp; &nbsp;at new Promise (&lt;anonymous&gt;)<br> &nbsp; &nbsp;at Object.bcrypt.compare (/server/node_modules/bcryptjs/dist/bcrypt.js:306:20)<br> &nbsp; &nbsp;at compare (/server/routes/sessions.js:23:30)<br> &nbsp; &nbsp;at &lt;anonymous&gt;<br> &nbsp; &nbsp;at process._tickCallback (internal/process/next_tick.js:188:7)</pre>
</body>
</html>
"
`;

exports[`Sessions Routes PUT /x/sessions update a session -- choose a subject 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Error: GlqI6VAROUCGqZy3OuKJ1A<br> &nbsp; &nbsp;at abort (/server/helpers/abort.js:4:22)<br> &nbsp; &nbsp;at abort (/server/middleware/authMiddleware.js:4:32)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/route.js:137:13)<br> &nbsp; &nbsp;at Route.dispatch (/server/node_modules/express/lib/router/route.js:112:3)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:281:22<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at Function.handle (/server/node_modules/express/lib/router/index.js:174:3)<br> &nbsp; &nbsp;at router (/server/node_modules/express/lib/router/index.js:47:12)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at /server/node_modules/body-parser/lib/read.js:130:5<br> &nbsp; &nbsp;at invokeCallback (/server/node_modules/raw-body/index.js:224:16)<br> &nbsp; &nbsp;at done (/server/node_modules/raw-body/index.js:213:7)<br> &nbsp; &nbsp;at IncomingMessage.onEnd (/server/node_modules/raw-body/index.js:273:7)<br> &nbsp; &nbsp;at emitNone (events.js:106:13)<br> &nbsp; &nbsp;at IncomingMessage.emit (events.js:208:7)<br> &nbsp; &nbsp;at endReadableNT (_stream_readable.js:1064:12)<br> &nbsp; &nbsp;at _combinedTickCallback (internal/process/next_tick.js:138:11)<br> &nbsp; &nbsp;at process._tickCallback (internal/process/next_tick.js:180:9)</pre>
</body>
</html>
"
`;

exports[`Sessions Routes PUT /x/sessions update a session -- choose a unit 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Error: GlqI6VAROUCGqZy3OuKJ1A<br> &nbsp; &nbsp;at abort (/server/helpers/abort.js:4:22)<br> &nbsp; &nbsp;at abort (/server/middleware/authMiddleware.js:4:32)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/route.js:137:13)<br> &nbsp; &nbsp;at Route.dispatch (/server/node_modules/express/lib/router/route.js:112:3)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:281:22<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at Function.handle (/server/node_modules/express/lib/router/index.js:174:3)<br> &nbsp; &nbsp;at router (/server/node_modules/express/lib/router/index.js:47:12)<br> &nbsp; &nbsp;at newFn (/server/node_modules/express-async-errors/index.js:25:20)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/server/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/server/node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /server/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/server/node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/server/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at /server/node_modules/body-parser/lib/read.js:130:5<br> &nbsp; &nbsp;at invokeCallback (/server/node_modules/raw-body/index.js:224:16)<br> &nbsp; &nbsp;at done (/server/node_modules/raw-body/index.js:213:7)<br> &nbsp; &nbsp;at IncomingMessage.onEnd (/server/node_modules/raw-body/index.js:273:7)<br> &nbsp; &nbsp;at emitNone (events.js:106:13)<br> &nbsp; &nbsp;at IncomingMessage.emit (events.js:208:7)<br> &nbsp; &nbsp;at endReadableNT (_stream_readable.js:1064:12)<br> &nbsp; &nbsp;at _combinedTickCallback (internal/process/next_tick.js:138:11)<br> &nbsp; &nbsp;at process._tickCallback (internal/process/next_tick.js:180:9)</pre>
</body>
</html>
"
`;
